packages/@aws-cdk/aws-events-targets/lib/apigateway.ts

```diff
@@ -0,0 +1,77 @@
+import * as apigateway from '@aws-cdk/aws-apigateway';
+import * as events from '@aws-cdk/aws-events';
+import * as iam from '@aws-cdk/aws-iam';
+import { singletonEventRole } from './util';
+
+/**
+ * Customize the API Gateway Event Target
+ */
+export interface ApiGatewayProps {
+  /**
+   * Method name
+   *
+   * @default `*`
+   */
+  readonly method?: string;
+
+  /**
+   * Resource path
+   *
+   * @default `*`
+   */
+  readonly path?: string;
+
+  /**
+   * Stage name
+   *
+   * @default `*`
+   */
+  readonly stage?: string;
+
+  /**
+   * Header, Path, and Querystring parameters
+   *
+   * @default - no parameters set
+   */
+  readonly httpParameters?: events.CfnRule.HttpParametersProperty;
```

Comment:
I don't believe we want to expose `Cfn*` types as inputs in the props as we don't want to leak that type (none of the other targets should accept them as props either IIRC).

It should be a part of the returned `RuleTargetConfig` but should not be an input